openpathsampling/engines/trajectory.py

```diff
@@ -284,6 +284,28 @@ def next(self):
 
         return ObjectIterator()
 
+    def index_of(self, sub_traj):
+        """
+        Return a list of indices of the snapshots in sub_traj within the trajectory
+
+        Parameters
+        ----------
+        sub_traj : :obj:`Trajectory`
+
+        Returns
+        -------
+        list of int or None
+            a list of integer indices or None, if a snapshot is not found in the trajectory
+        """
+        ret = []
+        for s in sub_traj:
+            if s in self:
+                ret.append(self.index(s))
+            else:
+                ret.append(None)
+
+        return ret
+
```

Comment:
I'm using it -- a lot -- in the rest of the analysis. It differs by taking a list of subtrajs.